components/server/ee/src/user/user-service.ts

```diff
@@ -30,10 +32,35 @@ export class UserServiceEE extends UserService {
 
         // the self-hosted case
         if (!this.licenseEvaluator.isEnabled(Feature.FeatureSetTimeout)) {
-            return "30m";
+            return WORKSPACE_TIMEOUT_DEFAULT_SHORT;
         }
 
-        return "60m";
+        return WORKSPACE_TIMEOUT_DEFAULT_LONG;
+    }
+
+    public workspaceTimeoutToDuration(timeout: WorkspaceTimeoutDuration): string {
+        switch (timeout) {
+            case WORKSPACE_TIMEOUT_DEFAULT_SHORT:
+                return "30m";
+            case WORKSPACE_TIMEOUT_DEFAULT_LONG:
+                return this.config.workspaceDefaults.timeoutDefault || "60m";
+            case WORKSPACE_TIMEOUT_EXTENDED:
+            case WORKSPACE_TIMEOUT_EXTENDED_ALT:
+                return this.config.workspaceDefaults.timeoutExtended || "180m";
+        }
+    }
+
+    public durationToWorkspaceTimeout(duration: string): WorkspaceTimeoutDuration {
+        switch (duration) {
+            case "30m":
+                return WORKSPACE_TIMEOUT_DEFAULT_SHORT;
+            case this.config.workspaceDefaults.timeoutDefault || "60m":
+                return WORKSPACE_TIMEOUT_DEFAULT_LONG;
+            case this.config.workspaceDefaults.timeoutExtended || "180m":
+                return WORKSPACE_TIMEOUT_EXTENDED_ALT;
+            default:
+                return WORKSPACE_TIMEOUT_DEFAULT_SHORT;
+        }
```

Comment:
I don't know if `user-service` is a good place for these functions. I was too miserly to create a dedicated service for this that does nothing but provide these values. However, I'm very open to proposals for better places.

Reply:
I also have no idea if `user-service` is a good place for this, but if nobody has an opinion on this, then let's say it is a good place. 🙌